backend/app/usecase/changelog/changelog.go

```diff
@@ -0,0 +1,55 @@
+package changelog
+
+import (
+	"github.com/short-d/app/fw"
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/keygen"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+// ChangeLog retrieves change log and create changes.
+type ChangeLog interface {
+	CreateChange(title string, summaryMarkdown string) (entity.Change, error)
+	GetChangeLog() ([]entity.Change, error)
+}
+
+// Persist retrieves change log from and saves changes to persistent data store.
+type Persist struct {
```

Comment:
@alldroll I am think that we will have billions of unique keys. This particular helps when we want to do database sharding without relying on each DB's own number incrementer ( which would have forced us to readjust all related DB records during partitioning process ). 

I think it's totally okay for changelog service to ensure the changes are persisted in the DB because its not write latency sensitive ( more storage sensitive ).

However, I do agree that for the real-time chat app, we can use message queue to decouple saving chat message from forwarding message to another client to reduce latency.

![image](https://user-images.githubusercontent.com/3537801/77675936-e052b080-6f4a-11ea-8358-afa65c41e6df.png)